src/v/pandaproxy/test/kafka_client_cache.cc

```diff
@@ -27,28 +27,29 @@
 
 using namespace std::chrono_literals;
 
+static constexpr auto clean_timer_period = 10s;
+
 namespace pandaproxy {
 struct test_client_cache : public kafka_client_cache {
-    ss::timer<ss::lowres_clock> clean_and_evict_timer;
-    ss::gate g;
-
     explicit test_client_cache(size_t max_size)
       : kafka_client_cache(
         to_yaml(kafka::client::configuration{}, config::redact_secrets::no),
         max_size,
-        1000ms,
-        std::reference_wrapper(clean_and_evict_timer)) {
-        clean_and_evict_timer.set_callback([this] {
-            ssx::spawn_with_gate(g, [this] { return clean_stale_clients(); });
-        });
-    }
+        1000ms) {}
+
+    test_client_cache(size_t max_size, std::chrono::milliseconds keep_alive)
+      : kafka_client_cache(
+        to_yaml(kafka::client::configuration{}, config::redact_secrets::no),
+        max_size,
+        keep_alive) {}
 };
 } // namespace pandaproxy
 
 namespace pp = pandaproxy;
 
 SEASTAR_THREAD_TEST_CASE(cache_make_client) {
     pp::test_client_cache client_cache{10};
+    client_cache.start().get();
```

Comment:
`client_cache.stop()` is never called. I'd be temped to add start.get() to the constructor of `test_client_cache` and `stop.get()` to the destructor.